### PR TITLE
@nivo/marimekko

### PR DESCRIPTION
**Pull request by plouc**

### CODE REVIEW

File: packages/marimekko/src/Bars.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { BarDatum, CommonProps, MouseEventHandlers } from './types'
+import { Bar } from './Bar'
+
+interface BarsProps<RawDatum> extends MouseEventHandlers<RawDatum, SVGRectElement> {
+    isInteractive: boolean
+    bars: BarDatum<RawDatum>[]
+    tooltip: CommonProps<RawDatum>['tooltip']
+}
+
+export const Bars = <RawDatum,>({
```
Comment (wyze) [inline packages/marimekko/src/Bars.tsx:13]: Same with the space here.

File: packages/marimekko/src/Marimekko.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import { Grid, Axes } from '@nivo/axes'
+import { InheritedColorConfig, OrdinalColorScaleConfig } from '@nivo/colors'
+import { BoxLegendSvg } from '@nivo/legends'
+import { SvgProps, LayerId, DimensionDatum } from './types'
+import { defaultProps } from './props'
+import { useMarimekko, useLayerContext, useLegendData } from './hooks'
+import { Bars } from './Bars'
+import { BarTooltip } from './BarTooltip'
+
+const InnerMarimekko = <RawDatum,>({
```
Comment (wyze) [inline packages/marimekko/src/Marimekko.tsx:18]: Same with the space.

File: packages/marimekko/src/Marimekko.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                    return <Fragment key={i}>{createElement(layer, layerContext)}</Fragment>
+                }
+
+                return null
+            })}
+        </SvgWrapper>
+    )
+}
+
+export const Marimekko = <RawDatum,>({
```
Comment (wyze) [inline packages/marimekko/src/Marimekko.tsx:186]: Same with the space.

File: packages/marimekko/src/ResponsiveMarimekko.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { ResponsiveWrapper } from '@nivo/core'
+import { Marimekko } from './Marimekko'
+import { SvgProps } from './types'
+
+export const ResponsiveMarimekko = <RawDatum,>(
```
Comment (wyze) [inline packages/marimekko/src/ResponsiveMarimekko.tsx:6]: Same with the space.

File: website/src/data/components/marimekko/meta.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    - isomorphic
+  stories:
+    - label: using arrays for data
+      link: marimekko--using-arrays-for-data
+    - label: diverging chart
+      link: marimekko--diverging
+    - label: custom layers
+      link: marimekko--custom-layers
+  description: |
+    The `Marimekko` component is somehow similar to a bar chart,
```
Comment (wyze) [inline website/src/data/components/marimekko/meta.yml:18]:
```suggestion
    The `Marimekko` component is somewhat similar to a bar chart,
```

File: website/src/data/components/marimekko/props.js [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+        type: 'string | (datum: RawDatum): string | number',
+        required: true,
+    },
+    {
+        key: 'value',
+        group: 'Base',
+        help: 'Value accessor.',
+        description: `
+            Define how to access the value of each datum,
+            wich is gonna dictate the thickness of the bars,
```
Comment (wyze) [inline website/src/data/components/marimekko/props.js:41]:
```suggestion
            which will dictate the thickness of the bars,
```

File: website/src/data/components/marimekko/props.js [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+            
+            The layer component which will receive the chart's
+            context & computed data and must return a valid SVG element
+            for the \`Marimekko\` component.
+            
+            The props passed to layers have the following structure:
+            
+            \`\`\`
+            {
+                data: ComputedData<RawDatum>[],
```
Comment (wyze) [inline website/src/data/components/marimekko/props.js:265]: I know there are a few extra props on the layer context, did you want to add them here?

**Review by wyze — CHANGES_REQUESTED** [review]
Overall, code looks great and performs as expected. Just some minor nits and questions.

File: packages/marimekko/src/Bar.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        height: number
+        opacity: number
+        color: string
+        borderColor: string
+    }>
+    isInteractive: boolean
+    tooltip: CommonProps<RawDatum>['tooltip']
+}
+
+export const Bar = <RawDatum,>({
```
[the remark this answers is not in the document]
Comment (wyze) [reply packages/marimekko/src/Bar.tsx:21]: Talked on discord about being a limitation of tooling before and now it works as expected.

File: packages/bullet/tests/Bullet.test.tsx [hunk 9]
```diff
@@ -257,7 +257,7 @@ describe('Bullet', () => {
                     "v1": 10,
                   },
                   "height": 80,
-                  "index": 504,
```
[the remark this answers is not in the document]
Comment (plouc) [reply packages/bullet/tests/Bullet.test.tsx:260]: I think so, looking at the previous code, the index is expected to be inherited from the data object where it's `0`, I couldn't figure out why we had `504`.

File: website/src/data/components/marimekko/props.js (hunk 7 above)
Comment (plouc) [reply website/src/data/components/marimekko/props.js:265]: Yes, good catch!

File: packages/annotations/package.json [hunk 10]
```diff
@@ -29,7 +29,7 @@
   "dependencies": {
     "@nivo/colors": "0.64.0",
     "lodash": "^4.17.11",
-    "react-spring": "^8.0.27"
+    "react-spring": "next"
```
[the remark this answers is not in the document]
Comment (plouc) [reply packages/annotations/package.json:32]: Yes, it might be safer.

File: website/src/data/components/marimekko/meta.yml (hunk 5 above)
Comment (plouc) [reply website/src/data/components/marimekko/meta.yml]: Fixed.

File: website/src/data/components/marimekko/props.js (hunk 6 above)
Comment (plouc) [reply website/src/data/components/marimekko/props.js]: Fixed.

File: website/src/data/components/marimekko/props.js (hunk 7 above)
Comment (plouc) [reply website/src/data/components/marimekko/props.js]: Added.

File: packages/annotations/package.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (plouc) [reply packages/annotations/package.json]: Changed for `9.0.0-rc.3`.

**Review by wyze — APPROVED** [review]
Looks good, thanks!

**Pull request merged**